contracts/clients/src/BlsWalletWrapper.ts

```diff
@@ -221,6 +227,41 @@ export default class BlsWalletWrapper {
     return await walletContract.nonce();
   }
 
+  /** Sign an operation with an estimate of the gas required. */
+  async signWithGasEstimate(
+    operation: Omit<Operation, "gas">,
+    overhead = 0,
+  ): Promise<Bundle> {
+    let gas = await this.estimateGas(operation);
+    gas = gas.add(gas.div(10000).mul(Math.ceil(overhead * 10000)));
```

Comment:
What are the `10000`'s here?

Reply:
@jacque006 This is approximating `gas + gas * overhead`. The reason to divide and multiply is to account for integer arithmetic - `overhead` is expected to be something like `0.1`, but we need to work with integers, so we multiply overhead by 10000 to get a whole number (after rounding) and compensate by dividing the other side of the product by 10000.

Reply:
Gotcha, that makes sense to me. Will this break/work sub-optimally with other overhead values on other L2 EVMs?

Reply:
@jacque006 I'm not sure what you mean. Could you give an example?

Reply:
Sorry, explained/thought through this poorly. Because of the fixed `10000` value you are using and the lack of rounding on the entire gas value, you can still get decimal values out of this arithmetic. I used the gas limit from https://explorer.arbitrum.io/tx/0xa0cd48e1f221e9ff876c441979b28852da93c3cebc49562dce1d654e4c4fa89e to test with a small overhead value:

![image](https://user-images.githubusercontent.com/10394161/216484245-4832a412-326a-4df4-a050-5c7a291ce9a4.png)

Reply:
> the lack of rounding on the entire gas value

Nah, `gas.div(10000)` is integer division, since `gas` is a `BigNumber`:

```
➜  contracts git:(main) node
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> ethers = require('ethers')
> ethers.BigNumber.from(15).div(7).toString()
'2'
```

Let's add a comment to clear this up.

Was that your only concern?

Reply:
Ah yes forgot about that, good is is then :+1: 

Comment would be good, test cases around a range of values would be better :)